src/Microsoft.DotNet.Build.Tasks/PackageFiles/BuildVersion.targets

```diff
@@ -0,0 +1,17 @@
+<Project xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+  
+  <PropertyGroup>
+    <TodayTimeStamp>$([System.DateTime]::Now.ToString(yyyyMMdd))</TodayTimeStamp>
+    <BuildVersionFile Condition="'$(BuildVersionFile)'==''">$(BaseIntermediateOutputPath)/$(TodayTimeStamp)/BuildVersion.props</BuildVersionFile>
+  </PropertyGroup>
+
+  <!-- If BuildValues.props exists already then import it to get BuildNumberMajor, else generate it and override props values. -->
+  <Import Condition="Exists('$(BuildVersionFile)')" Project="$(BuildVersionFile)" />
+
+  <PropertyGroup Condition="!Exists('$(BuildVersionFile)')">
+    <GenerateAssemblyInfoDependsOn>CreateVersionFileDuringBuild;$(GenerateAssemblyInfoDependsOn)</GenerateAssemblyInfoDependsOn>
+    <GetNuGetPackageDependenciesDependsOn>CreateVersionFileDuringBuild;$(GetNuGetPackageDependenciesDependsOn)</GetNuGetPackageDependenciesDependsOn>
```

Comment:
We should avoid mutating these dependsons from various other targets files here. In stead we should provide a common target they can depend on to get the build numbers and we do different things in that target based one the different modes we have.